keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -638,24 +640,153 @@ always dispatched immediately.  Following is how the other requests
 are dispatched at a given apiserver.
 
 The concurrency limit of an apiserver is divided among the non-exempt
-priority levels in proportion to their assured concurrency shares.
-This produces the assured concurrency value (ACV) for each non-exempt
-priority level:
+priority levels, and higher ones can do a limited amount of borrowing
+from lower ones.
 
-```
-ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+Two fields of `LimitedPriorityLevelConfiguration`, introduced in the
+midst of the `v1beta2` lifetime, configure the borrowing.  The following
+display shows the two new fields along with the updated description for
+the `AssuredConcurrencyShares` field.
+
+```go
+type LimitedPriorityLevelConfiguration struct {
+  ...
+  // `assuredConcurrencyShares` (ACS) contributes to the computation of the
+  // NominalConcurrencyLimit (NCL) of this level.
+  // This is the number of execution seats available at this priority level.
+  // This is used both for requests dispatched from
+  // this priority level as well as requests dispatched from higher priority
+  // levels borrowing seats from this level.  This does not limit dispatching from
+  // this priority level that borrows seats from lower priority levels (those lower
+  // levels do that).  The server's concurrency limit (SCL) is divided among the
+  // Limited priority levels in proportion to their ACS values:
+  //
+  // NCL(i)  = ceil( SCL * ACS(i) / sum_acs )
+  // sum_acs = sum[limited priority level k] ACS(k)
+  //
+  // Bigger numbers mean a larger nominal concurrency limit, at the expense
+  // of every other Limited priority level.
+  // This field has a default value of 30.
+  // +optional
+  AssuredConcurrencyShares int32
+
+  // `lendableConcurrencyShares` (LCS) contributes to the computation of the
+  // LendableConcurrencyLimit (LCL) for this level.  This is the number of
+  // execution seats of this level that can be borrowed by higher priority
+  // Limited levels.
+  // This may not be negative, and may not be greater than
+  // `assuredConcurrencyShares`.
+  //
+  // LCL(i) = ceil( SCL * LCS(i) / sum_acs )
+  //
+  // This field has a default value of zero.
+  // +optional
+  LendableConcurrencyShares int32
```

Comment:
The other way one could present this is as a percentage (of ACS), that would make it even clearer that it's not additive, and also if you change ACS you wouldn't have to update this field to keep the same proportion.

The downside is that maybe people wouldn't want to keep the same proportion most of the time; I can't yet guess which would be most common.

Reply:
(but see subsequent comment for yet a different thought)

Reply:
I agree, we could use a fraction or percentage here rather than another share.  That might indeed make the design clearer.

I do not like an integer percentage, because it makes an arbitrary limit on granularity.  OTOH, there is the first bullet in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types .  So shares avoids both of those problems.

Reply:
Switched to percentage.